micropython/senml/manifest.py

```diff
@@ -1,5 +1,7 @@
-metadata(version="0.1.0")
+metadata(
+    version="0.1.0", description="SenML serialisation for MicroPython.", pypi="micropython-senml"
+)
 
-require("cbor2")
+require("cbor2", pypi="cbor2")
```

Comment:
This kwarg would need to be repeated each time this package is required.... maybe it would be better to instead have the target package define what it's PyPI package name is, eg in the metadata?  That's more self-documenting ("this package is intended to implement a subset of ...") and has less duplication.

OTOH, maybe it is better to have this dependency written out here, because maybe the package requiring it needs more control over what it depends on.  Eg there may be two packages in micropython-lib:
1. senml, requires cbor2 (micropython-lib) and cbor2 (PyPI)
2. foobar, requires cbor2 (micropython-lib) and foobor2 (PyPI)

??

Reply:
> OTOH, maybe it is better to have this dependency written out here, because maybe the package requiring it needs more control over what it depends on.

Yes, this is exactly the intention. I started with the former, with exactly the same idea "it would be neat if cbor2 knew it was actually based on PyPI/cbor2", but I think in general you want the dependee to specify its dependencies explicitly. (This also includes the ability to pin a version).

Reply:
OK, very good.

I guess the fact that something is in `python-ecosys` with a given name, means it's supposed to be based on a PyPI package.

Maybe one day we can add a field to `metadata` to make this explicit.  Or just include the PyPI name in the description.